hbase_sql/src/main/java/com/splicemachine/ddl/ZooKeeperDDLWatchChecker.java

```diff
@@ -67,11 +76,32 @@ public void process(WatchedEvent watchedEvent){
                 }
             }
         };
-
-
+        initDemarcationPoint();
         return true;
     }
 
+    private void initDemarcationPoint() throws IOException{
+        String rootPath = SIDriver.driver().getConfiguration().getSpliceRootPath();
+        String ddlPath = rootPath+HConfiguration.DDL_PATH;
+        List<String> children = ZkUtils.getChildren(ddlPath, null);
+        long max = 0;
+        for (String child : children) {
+            String demarcationPath = ddlPath + "/" + child;
+            byte[] data = ZkUtils.getData(demarcationPath);
+            long demarcationPoint = data != null ? Bytes.toLong(data) : 0;
+            if (demarcationPoint > max) {
+                max = demarcationPoint;
+            }
+        }
+        String demarcationPointPath = ddlPath+"/"+id;
+        try {
+            ZkUtils.recursiveSafeCreate(demarcationPointPath, Bytes.toBytes(max), ZooDefs.Ids.OPEN_ACL_UNSAFE,
+                    CreateMode.EPHEMERAL);
+            SpliceLogUtils.info(LOG, "Initialize DDL demarcation point to %d", max);
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+    }
```

Comment:
Rather than storing it back into ZK, we can store it locally and return it in `getDemarcationPoint()`

Reply:
Ok, I think that makes sense.